hydra-node/src/Hydra/Chain/Direct/Handlers.hs

```diff
@@ -230,21 +231,19 @@ chainSyncHandler tracer callback getTimeHandle ctx =
                       }
                 }
 
--- | Hardcoded grace time for close transaction to be valid.
--- TODO: make it a node configuration parameter
-closeGraceTime :: SlotNo
-closeGraceTime = 100
-
-fromPostChainTx ::
+prepareTxToPost ::
   (MonadSTM m, MonadThrow (STM m)) =>
   TimeHandle ->
   TinyWallet m ->
   ChainContext ->
   ChainStateType Tx ->
   PostChainTx Tx ->
   STM m Tx
-fromPostChainTx timeHandle wallet ctx cst@ChainStateAt{chainState} tx = do
-  pointInTime <- throwLeft currentPointInTime
+prepareTxToPost timeHandle wallet ctx cst@ChainStateAt{chainState} tx = do
+  pointInTime@(currentSlot, _) <- throwLeft currentPointInTime
```

Comment:
This will fail on a real network! Using the `currentSlot` (as it is currently computed by `queryTimeHandle` - this is where `currentPointInTime` is coming from) will be rejected by a cardano-node that has not seen a block with a slot `>` than this one in it!

Must: Change `queryTimeHandle` to actually use the last known slot (== tip) as reported by the node as `currentSlot`. Also, check whether we rely on the `currentSlot` being the projected, wall clock slot -> we would break this code when we change the semantics of `queryTimeHandle`.